### PR TITLE
Fixed exception when downloading credit slip on an order with a deleted product

### LINKED ISSUE #29495: BO - credit-slips - An exception is displayed when downloading a credit slip with deleted product
## Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

## Describe the bug and add screenshots

In the BO > Orders > Credit slips page on PrestaShop 8.0.0, an exception is displayed when downloading a credit slip with a deleted product

image

_Important: I didn't reproduce the issue with php7.2, thanks @hibatallahAouadni_

## Expected behavior

The expected behavior is the behavior from PrestaShop 1.7.8.7

I checked with PrestaShop 1787 with php7.2 : behavior is correct.

### If the credit slip has only one product that has already been deleted from the order details, this is the display
image

### If the credit has more than one product

####Before deleting the product
image

#### Then if you delete a product from the order details it now looks like this
image

## Steps to reproduce

1. Enable debug mode
2. Create an order from the BO or FO
3. Go to backoffice page BO > Orders > View to look at the order you just created
4. Change the order status to 'payment: accepted', this validates the order
5. Click on 'Partial refund' button and use the interface to 'return the product': this simulates an order where a user is not satisfied with product and returns the sold product, and prestashop creates a refund
6. From the order page, **remove the product** from the order details page 

image

7. Go to backoffice page BO > Orders > Credit slips page
8. Click on "Download Credit slip" for the order
9. See error: exception message

image

10. Click on "Generate PDF file by date"
11. See exception 

## PrestaShop version(s) where the bug happened

8.0.0beta, 1787

### PHP version(s) where the bug happened

7.4, 8.0

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x 
| Description?      | 
| Type?             | bug fix 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29495
| Related PRs       | N/A
| How to test?      | Please see https://github.com/PrestaShop/PrestaShop/issues/29495
| Possible impacts? |

### CODE REVIEW

File: classes/order/OrderSlip.php [hunk 1]
```diff
@@ -508,18 +508,20 @@ public function getEcoTaxTaxesBreakdown()
                 WHERE `id_order_detail` = ' . (int) $order_slip_details['id_order_detail']
             );
 
-            if (!isset($ecotax_detail[$row['rate']])) {
-                $ecotax_detail[$row['rate']] = ['ecotax_tax_incl' => 0, 'ecotax_tax_excl' => 0, 'rate' => $row['rate']];
-            }
+            if (false !== is_array($row)) {
```
Comment (matks) [inline classes/order/OrderSlip.php]:
```suggestion
            if (true === is_array($row)) {
```
I think it's a little easier to read but it's just my opinion. Do you think `false !==` is better? 😄

**Review by matks — DISMISSED** [review]
Code seems valid => I just added a nice-to-have item

File: classes/order/OrderSlip.php (hunk 1 above)
Comment (nicosomb) [reply classes/order/OrderSlip.php]: 💯

**Review by atomiix — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

File: classes/order/OrderSlip.php (hunk 1 above)
Comment (yanmakouf) [reply classes/order/OrderSlip.php]: ying yang

**Review by HanaRebaiQA — APPROVED** [review]
Hello @nicosomb 

I have checked this PR. The issue is fixed.

Here is my checked list : 

- Check when creating Order from BO : OK
- Check when creating Order from FO : OK
- Check when add to cart with one product : OK
- Check when add to cart with various products : OK
- Check with multistore : OK
- Check with different languages (RTL and No RTL) : OK
- Check with different php versions (php7.3 and php8.0) : OK
- Check the automatic tests : OK

image



https://user-images.githubusercontent.com/92912932/192578418-64961eab-8e07-4ac5-9166-ec7fafc6caf1.mp4


https://user-images.githubusercontent.com/92912932/192578431-16ddb0b1-9d1f-4cad-bf73-517857cb9557.mp4

So, it's QA :heavy_check_mark: 

Thank You!

**Pull request merged**